module/zfs/vdev.c

```diff
@@ -1347,6 +1347,10 @@ vdev_open(vdev_t *vd)
 		vd->vdev_max_asize = max_asize;
 		if (vd->vdev_ashift == 0)
 			vd->vdev_ashift = ashift;
+		else
+			if (vd->vdev_ashift < ASHIFT_MIN ||
+			    vd->vdev_ashift > ASHIFT_MAX)
```

Comment:
You can add `VDEV_AUX_BAD_ASHIFT` as an entry at the end of the `vdev_aux_t` enum. This should result in the disk being labelled as `UNAVAIL` which I think is appropriate. The fact that this case also returns `EDOM` provides even more logging. I would still check upstream to make sure the enum hasn't changed as well, but we've pulled in almost all of the upstream patches and it hasn't changed thus far.

@behlendorf any other thoughts?